schema.d.ts

```diff
@@ -138,16 +138,30 @@ export type Declaration =
  *
  * All references are required to be publically accessible, so the canonical
  * representation of a reference is the export it's available from.
- *
- * Referrences to global symbols like `Array`, `HTMLElement`, or `Event`
- *
+ * 
+ * `package` should generally refer to an npm package name. If `package` is
+ * undefined then the reference is local to this package. If `module` is
+ * undefined the reference is local to the containing module.
+ * 
+ * References to global symbols like `Array`, `HTMLElement`, or `Event` should
+ * use a `package` name of `"global:"`.
```

Comment:
How is this better than just omitting a `package` or `module` for globals? My worry here is that this starts to feel like a custom type system (what does the `:` after `"global"` mean?)

Reply:
This is just namespacing, not types. Omitting the package means that the reference is inside the manifest's package. If no package meant global, then we'd just have to find a namespace for "this package" as well. The `:` is just an invalid package name character to not collide with https://www.npmjs.com/package/global

Reply:
I still dont really understand this. Maybe can you provide a practical example? What does this look like in source code, and what is the expected output?

Reply:
This would be a reference to the global `Array`:

```json
"type": {
  "text": "Array<number>",
  "references": [
    {
      "start": 0,
      "end": 5,
      "package": "global:",
      "name": "Array"
    }
  ]
}
```